lib/drawSquareDoubles.c

```diff
@@ -14,6 +13,95 @@
 #define DEBUG2 '2'
 #define DEBUG3 '3'
 
+triangle* meshInit (int numberOfTris){
+    return (triangle*) malloc(numberOfTris * sizeof(triangle));
+}
+
+void plotTrianglePoints(double arr[][3], triangle* triangles, int numberOfTris){
+
+    int totalNumberOfPoints = numberOfTris*3;
+    int count = 0;
+
+    for (int i = 0; i < numberOfTris; i++){
+        for(int j = 0; j < 3; j++){
+                
+            arr[count][0] = triangles[i].p[j].x;
+            arr[count][1] = triangles[i].p[j].y;
+            arr[count][2] = triangles[i].p[j].z;
+            count++;    
+        }
+    }
+}
+
+void projectTrianglePoints2d(double arr[][3], double p_points[][2], const double DISTANCE, int iter){
+        for(int i = 0; i < iter; i++){
+            //extract x, y and z from array.
+
+            double x = arr[i][0];
+            double y = arr[i][1];
+            double z = arr[i][2];
+
+            double zPerspective = 1/(DISTANCE - z);
+
+            double p_Mat[2][3] = {{zPerspective,0,0},{0,zPerspective,0}};
+            
+            double x_p = (p_Mat[0][0]*x)+(p_Mat[0][1]*y)+(p_Mat[0][2]*z);
+            double y_p = (p_Mat[1][0]*x)+(p_Mat[1][1]*y)+(p_Mat[1][2]*z);
+            
+            p_points[i][0] = x_p;
+            p_points[i][1] = y_p;
+        }
+}
+
+void drawTriangleOnScreen(double arr[][2],double origin[2], double ratio, int screen[MAX_X][MAX_Y], int iter, int totaltris){
```

Comment:
is there any special logic in here pertaining to triangles, or this idential to previous line drawing?

Reply:
Same reason as above, started as a duplicate function before I really knew what it was going to do. Function name not really desciptive.